packages/benchmarks/src/textToDriver/fuzzyMatch.test.ts

```diff
@@ -0,0 +1,403 @@
+
+    const expected = JSON.stringify([
+      { id: 1, name: "Alice" },
+      { id: 3, name: "Charlie" },
+    ]);
+
+    const result = fuzzyMatch({ mongoDbOutput, expected });
+
+    expect(result).toBe(false);
+  });
+});
+
+describe("fuzzyMatch() with Aggregations", () => {
+  test("should return true for exact matching aggregation with a single document", () => {
+    const mongoDbOutput = [{ count: 10 }];
+    const expected = JSON.stringify([{ count: 10 }]);
+
+    const result = fuzzyMatch({
+      mongoDbOutput,
+      expected,
+      isAggregation: true,
+    });
+
+    expect(result).toBe(true);
+  });
+  test("should return true for fuzzy match aggregation with multiple documents", () => {
+    const mongoDbOutput = [{ count: 1 }, { count: 2 }];
+
+    const expected = JSON.stringify([{ OTHER: 1 }, { OTHER: 2 }]);
+
+    expect(
+      fuzzyMatch({
+        mongoDbOutput,
+        expected,
+        isAggregation: true,
+      })
+    ).toBe(true);
+  });
+
+  test("should return true when scalar aggregation result matches with allowed tolerance", () => {
+    const allowedNumberDifference = 0.05;
+
+    const mongoDbOutput = { count: 10 + allowedNumberDifference / 2 };
+    const expected = JSON.stringify([{ count: 10 }]);
+
+    const result = fuzzyMatch({
+      mongoDbOutput,
+      expected,
+      isAggregation: true,
+      allowedNumberDifference,
+    });
+
+    expect(result).toBe(true);
+  });
+
+  test("should return false when scalar aggregation result exceeds allowed tolerance", () => {
+    const allowedNumberDifference = 0.05;
+    const mongoDbOutput = { count: 11 + allowedNumberDifference * 2 };
+    const expected = JSON.stringify([{ count: 10 }]);
+
+    const result = fuzzyMatch({
+      mongoDbOutput,
+      expected,
+      isAggregation: true,
+      allowedNumberDifference,
+    });
+
+    expect(result).toBe(false);
+  });
+
+  test("should return true when aggregation result is a scalar and matches", () => {
+    const mongoDbOutput = 100;
+    const expected = JSON.stringify([{ "count(*)": 100 }]);
+
+    const result = fuzzyMatch({
+      mongoDbOutput,
+      expected,
+      isAggregation: true,
+    });
+
+    expect(result).toBe(true);
+  });
+  test("should return false when aggregation result is a scalar and does not match", () => {
+    const mongoDbOutput = 50;
+    const expected = JSON.stringify([{ "count(*)": 100 }]);
+
+    const result = fuzzyMatch({
+      mongoDbOutput,
+      expected,
+      isAggregation: true,
+    });
+
+    expect(result).toBe(false);
+  });
+
+  test("should return true when aggregation output has extra keys", () => {
+    const mongoDbOutput = [{ count: 10, extra: "ignore" }];
+    const expected = JSON.stringify([{ count: 10 }]);
+
+    const result = fuzzyMatch({
+      mongoDbOutput,
+      expected,
+      isAggregation: true,
+    });
+
+    expect(result).toBe(true);
+  });
+
+  test("should return true when aggregation output has mismatched keys", () => {
```

Comment:
This is an interesting case! Is this the idea behind the `isAggregation` flag in general?

i.e. "We know the result will be `10`. Ideally it's in the `total` field but we actually don't _really_ care what the field name is for fuzzy matching."

Reply:
yeah exactly